Homework18

OneAndOnlyAdam opened the pull request:
First time running via Terminal in IntelliJ:  Shows build successful so I guess that's good.

OneAndOnlyAdam committed:
Homework18

First time running via Terminal in IntelliJ:  Shows build successful so I guess that's good.

Berichicko commented:
Please finish your homework.

**The browser does not open in Homework18**

HINT

**Step 1**

**BaseTest** — Implements the necessary methods for work with webdriver.

The BaseTest class contains common setup and teardown methods, as well as any shared functionality or configurations required by multiple test classes.


**@BeforeSuite**	- The annotated method will be run before all tests in this suite have run.

**@BeforeMethod** -	The annotated method will be run before all the test methods in the current class have been run.

**@AfterMethod**	- The annotated method will be run after each test method.

_For example_

![image](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/a72d1f62-e2a4-4e74-b7bb-866bbf84523e)

```
    @BeforeMethod
    public void launchBrowser() {
        // Create ChromeOptions object to configure ChromeDriver
        ChromeOptions options = new ChromeOptions();

        // Add arguments to the ChromeOptions object
        options.addArguments("--disable-notifications","--remote-allow-origins=*", "--incognito","--start-maximized");

        // Set experimental options to exclude certain switches
        options.setExperimentalOption("excludeSwitches", new String[]{"enable-automation"});

        // Create a new instance of ChromeDriver with the configured options
        driver = new ChromeDriver(options);

        // Set the implicit wait time for the driver
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));

        // Open the specified URL in the Chrome browser
        driver.get("https://qa.koel.app/");
    }


    @AfterMethod
    public void closeBrowser() {
        // Quit the driver, closing all associated windows
        driver.quit();
    }
```

**Step 2**

 The <methods> element should be a child of the <class> element. Here's the corrected version:


```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">

<suite name="config">

    <test name="Play Song" preserve-order="false">
        <classes>
            <class name="Homework18"></class>
        </classes>
    </test>
</suite>
```

![image](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/28e1c50d-6c29-4067-b56c-879e3185a7fb)

**Step 3**

![18](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/c3df71e7-33f4-4a3e-a581-2ca71fb5c566)

OneAndOnlyAdam committed:
Still failing?

I'm not sure what is wrong.  When I run it through the terminal: 'gradle clean test' it fails.  When. i click on the report I don't understand what the problem is???

Berichicko commented:
HINT

**1)**

No comments in code

For examle

```
test {
    useTestNG() {
       suites "src/test/resources/config.xml"
    }
}
```

![image](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/b5773672-b18f-42e5-acce-5d54889504b3)


**2)**

![image](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/9176fa8a-a480-4849-accd-aa94509f7bca)


```
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.testng.Assert;
import org.testng.annotations.Test;

public class Homework18 extends BaseTest {
    @Test(priority = 1, description = "Play song and validate is is playing")
    public void playSong() throws InterruptedException{
        logIn("adam.johnson@testpro.io", "1Te$t$tudent");

        enterAllSongs();
        selectSong();
        enterButtonPlaySong();


        Assert.assertTrue(isDisplayedPlayingSong());

    }



    public boolean isDisplayedPlayingSong(){;
        WebElement songIsPLaying = driver.findElement(By.cssSelector("[data-testid = 'sound-bar-play']"));
        return songIsPLaying.isDisplayed();
    }
}
```

And

```
 public void enterAllSongs() throws InterruptedException{
        WebElement allSongs = driver.findElement(By.cssSelector("a[href='#!/songs']"));
        allSongs.click();
        Thread.sleep(5000);
    }
    
    public void selectSong() throws InterruptedException{
        WebElement song = driver.findElement(By.xpath("//tr[@class='song-item']"));
        song.click();
        Thread.sleep(1000);
    }

    public void enterButtonPlaySong() {
        WebElement buttonPlaySong = driver.findElement(By.xpath("//span[@title='Play or resume']"));
        Actions actions = new Actions(driver);
        actions.click(buttonPlaySong).perform();
    }
```
![image](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/15e78e5e-31a1-442a-90ac-492c86a2e3fc)

![image](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/83ed2aeb-1ebc-41fb-8acc-8fb5c6b972c2)
